tests/system/model_monitoring/test_app.py

```diff
@@ -1122,15 +1125,18 @@ def _log_model(self) -> None:
 
     def _deploy_monitoring_app(self) -> None:
         self.project = typing.cast(mlrun.projects.MlrunProject, self.project)
+        # create temp dir in cwd
+        Path.cwd().joinpath("temp").mkdir(exist_ok=True)
```

Comment:
Use `with` construct and the tempfile package (instead of shutil) to make sure the temp-dir gets deleted also on exceptions.
Something like
```
        with tempfile.TemporaryDirectory(dir=Path.cwd()) as temp_dir:
            fn = self.project.set_model_monitoring_function(
                func=self.app_data["url"],
                application_class=self.app_data["class_name"],
                name=self.app_data["app_name"],
                image="mlrun/mlrun" if self.image is None else self.image,
                local_path=temp_dir,
            )
            self.project.deploy_function(fn)
            return fn
```